packages/webapp-server/src/controllers/kErrorController.ts

```diff
@@ -0,0 +1,130 @@
+import { NextFunction, Request, Response } from 'express';
+import { ClusterModel } from '../models/ClusterModel';
+import { NativeKEvent, KErrorModel } from '../models/KErrorModel';
+
+export interface KErr {
+  name: string;
+  reason: string;
+  message: string;
+  type: string;
+  firstTimestamp: string;
+  lastTimestamp: string;
+  count: number;
+  nativeEvent: NativeKEvent;
+}
+
+export const kErrorController = {
+  getCluster: async (req: Request, res: Response, next: NextFunction) => {
+    const clusterId = req.headers['clientId'] as string;
+    const clusterSecret = req.headers['clientSecret'] as string;
+
```

Comment:
I added this middleware temporarily, but this logic should probably live inside the cluster controller

Reply:
Althought his particular middleware could be used in place of a user since it checks for a secret, but others would verify the user is logged in so no secret is needed (just have to check that the user is either an owner or member of that cluster)